tests/unit/download/test_api.py

```diff
@@ -0,0 +1,297 @@
+from macrosynergy.download import dataquery
+from macrosynergy.download import JPMaQSDownload
+from typing import List
+from unittest import mock
+from random import random
+import unittest
+import numpy as np
+
+class TestCertAuth(unittest.TestCase):
+    def test_something(self):
+        self.assertEqual(True, True)  # add assertion here
+
+
+class TestOAuth(unittest.TestCase):
+    def test_init(self):
+        oauth = dataquery.OAuth(client_id="test-id", client_secret="SECRET")
+
+        self.assertEqual(dataquery.OAUTH_BASE_URL, oauth.base_url)
+        self.assertEqual("test-id", oauth.client_id)
+        self.assertEqual("SECRET", oauth.client_secret)
+
+        self.assertIsNone(oauth.status_code)
+        self.assertIsNone(oauth.last_response)
+        self.assertIsNone(oauth.last_url)
+
+    def test_invalid_dtype_client_id(self):
+        with self.assertRaises(AssertionError):
+            dataquery.OAuth(client_id=b"test-id", client_secret="SECRET")
+
+    def test_invalid_dtype_client_secret(self):
+        with self.assertRaises(AssertionError):
+            dataquery.OAuth(client_id="test-id", client_secret=b"SECRET")
+
+    def test_valid_token(self):
+        oauth = dataquery.OAuth(client_id="test-id", client_secret="SECRET")
+        self.assertFalse(oauth._valid_token())
```

Comment:
Won't this make an actually call to DQ? And if so, shouldn't it be an integration test instead of unit test?

Reply:
OAuth._valid_token does not make a call to DQ; it only checks the state of the stored token (if there is one. if not, it returns false)